src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,107 @@
+package duke.parser;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.exception.CommandNotFoundException;
+import duke.exception.DukeException;
+import duke.exception.InvalidDateTimeException;
+
+
+/**
+ * A class represents a Parser.
+ */
+public class Parser {
+    /**
+     * Returns a Command based on user's input.
+     * @param input The full command from user's input.
+     * @return A command base on the input.
+     * @throws DukeException If error occurs during the process.
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        int endIndex = input.indexOf(" ");
+        String potentialCommand = (endIndex == -1) ? input : input.substring(0, endIndex);
+        Command command;
+
+        if (potentialCommand.equalsIgnoreCase("TODO")) {
+            command = new TodoCommand(input);
+        } else if (potentialCommand.equalsIgnoreCase("DEADLINE")) {
+            command = new DeadlineCommand(input);
+        } else if (potentialCommand.equalsIgnoreCase("EVENT")) {
+            command = new EventCommand(input);
+        } else if (potentialCommand.equalsIgnoreCase("LIST")) {
+            command = new ListCommand();
+        } else if (potentialCommand.equalsIgnoreCase("DONE")) {
+            command = new DoneCommand(input);
+        } else if (potentialCommand.equalsIgnoreCase("BYE")) {
+            command = new ExitCommand();
+        } else if (potentialCommand.equalsIgnoreCase("DELETE")) {
+            command = new DeleteCommand(input);
+        } else if (potentialCommand.equalsIgnoreCase("FIND")) {
+            command = new FindCommand(input);
+        } else {
+            throw new CommandNotFoundException("What do you mean? I do not know this command.");
+        }
+        return command;
+    }
+
+    /**
+     * Returns a LocalDateTime from a string containing date and time.
```

Comment:
It's not immediately clear how the datetime is parsed.
You might want to add in a description of how this parsing is done as well, as an
additional paragraph under the method summary.

Reply:
Will add and many thanks!